itest/rust/src/object_tests/class_rename_test.rs

```diff
@@ -0,0 +1,33 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use crate::framework::itest;
+use godot::prelude::*;
+
+pub mod dont_rename {
+    use super::*;
+
+    #[derive(GodotClass)]
+    #[class(init, base = Node)]
+    pub struct RepeatMe {}
+}
+
+pub mod rename {
+    use super::*;
+
+    #[derive(GodotClass)]
+    #[class(init, base = Node, rename = NoRepeat)]
+    pub struct RepeatMe {}
+}
+
+#[itest]
+fn renaming_changes_the_name() {
+    // note: this test doesn't really fail; if it doesn't succeed, then
+    // the runner hangs.  I don't know how to solve that.
+    assert!(dont_rename::RepeatMe::class_name() != rename::RepeatMe::class_name());
+    assert_eq!(dont_rename::RepeatMe::class_name().as_str(), "RepeatMe");
+    assert_eq!(rename::RepeatMe::class_name().as_str(), "NoRepeat");
+}
```

Comment:
Just hangs -- do you see any error or so?

First statement could be `assert_ne!`, by the way 🙂

Reply:
Hang is maybe imprecise?  If you don't rename the class, you hit the error mentioned in the issue and Godot throws some errors and the runner can't proceed

```
> godot4 --path itest/godot --headless -- \[\]
Initialize GDExtension API for Rust: Godot Engine v4.1.1.stable.official
Godot Engine v4.1.1.stable.official.bd6af8e0e - https://godotengine.org
 
ERROR: Rust function panicked in file godot-core/src/registry.rs at line 209. Context: failed to initialize GDExtension level `Scene`
   at: <function unset> (godot-core/src/lib.rs:154)
ERROR: Panic msg:  the definition for `RepeatMe` is overloaded; you may have multiple `GodotClass`es named `RepeatMe` which you should #[class(rename=...)]
   at: <function unset> (godot-core/src/lib.rs:102)
SCRIPT ERROR: Parse Error: Identifier "IntegrationTests" not declared in the current scope.
          at: GDScript::reload (res://TestRunner.gd:31)
```

Reply:
But I've now discovered that this is solved by not having the `init` or `base` in the class declarations as you suggested.

Reply:
I always forget about `assert_ne`, I always try `assert_neq` and when that doesn't work, go for `assert`